Mathlib/Data/List/Infix.lean

```diff
@@ -210,6 +216,19 @@ theorem prefix_iff_eq_take : l₁ <+: l₂ ↔ l₁ = take (length l₁) l₂ :=
     fun e => e.symm ▸ take_prefix _ _⟩
 #align list.prefix_iff_eq_take List.prefix_iff_eq_take
 
+theorem prefix_take_iff {x y : List α} {n : ℕ}: x <+: y.take n ↔ x <+: y ∧ x.length ≤ n := by
+  constructor
+  { intro h; constructor
+    exact List.IsPrefix.trans h (List.take_prefix _ _)
+    replace h := h.length_le; simp[Nat.le_min] at h; exact h.left }
+  { intro ⟨hp, hl⟩; have hl' := hp.length_le; rw[List.prefix_iff_eq_take] at *
+    rw[hp, List.take_take]; simp[min_eq_left, hl, hl'] }
+theorem concat_prefix_of_length_lt {x y : List α} (h : x <+: y) (hl : x.length < y.length) :
+    x ++ [y.get ⟨x.length, hl⟩] <+: y := by
+  use y.drop (x.length + 1); dsimp; nth_rw 1[List.prefix_iff_eq_take.mp h]
```

Comment:
Similarly, the `dsimp` is non-terminal. Please "squeeze" it also, i.e. replace by a `dsimp only` call.